File: clients/python/examples/tables-alter-type-rename.py
from astrapy import DataAPIClient
from astrapy.info import (
    AlterTypeRenameFields,
)

# Get a database
client = DataAPIClient()
database = client.get_database(
    "**API_ENDPOINT**", token="**APPLICATION_TOKEN**"
)

# Rename fields in a user-defined type
database.alter_type(
    "member",
    AlterTypeRenameFields(
        fields={"name": "first_name", "is_active": "is_member"}
    ),
)

File: clients/java/examples/tables-alter-type-rename.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.databases.Database;
import com.datastax.astra.client.tables.commands.AlterTypeRenameFields;

public class Example {

  public static void main(String[] args) {
    // Get a database
    Database database = new DataAPIClient("**APPLICATION_TOKEN**").getDatabase("**API_ENDPOINT**");

    // Rename fields in a user-defined type
    database.alterType(
        "member",
        new AlterTypeRenameFields()
            .addField("name", "first_name")
            .addField("is_active", "is_member"));
  }
}
